lib/errors.js

```diff
@@ -185,4 +185,29 @@ errors.GatewayTimeoutError = function(message) {
 };
 util.inherits(errors.GatewayTimeoutError, errors.SmartcarError);
 
+/**
+ * Enhanced errors from API v2.0
+ *
+ * @param {Object|String} error - response body from a v2.0 request
+ */
+class SmartcarErrorV2 extends errors.SmartcarError {
+  constructor(error) {
+    super();
+    if (error instanceof String) {
+      this.message = error;
+      return;
+    }
+    // standard v2 error attributes
+    this.type = error.type;
+    this.code = error.code;
+    this.description = error.description;
+    this.statusCode = error.statusCode;
+    this.requestId = error.requestId;
+    this.resolution = error.resolution;
+    this.docURL = error.docURL;
+    this.detail = error.detail;
```

Comment:
let's document these properties for the JSDoc

https://stackoverflow.com/questions/51467835/how-to-use-jsdoc-to-document-an-es6-class-property